ImageAnalysis/image_analysis/utils.py

```diff
@@ -118,13 +118,23 @@ def get_magspecstitcher_timestamp(fname):
     return timestamp_seconds
 
 def get_picoscopeV2_timestamp(file_path):
-    # Load the TDMS file
-    tdms_file = TdmsFile.read(file_path)
-    
-    # Access the file properties
-    timestamp = tdms_file.properties.get('timestamp', None)
-    
-    return timestamp
+    """
+    Retrieve the 'timestamp' property from a TDMS file efficiently.
+
+    Args:
+        file_path (str or Path): Path to the TDMS file.
+
+    Returns:
+        The timestamp if it exists, or None otherwise.
+    """
+    try:
+        # Open the TDMS file in lazy mode (if supported by the library)
+        with TdmsFile.open(file_path) as tdms_file:
+            return tdms_file.properties.get('timestamp')
+    except Exception as e:
```

Comment:
Be more specific with which Exception you're catching - i.e. only catch the Exception that you expect from file reading. You want any unexpected Exceptions to fail loudly.